src/scheduler.c

```diff
@@ -66,6 +69,16 @@ int main(int argc, char *argv[])
             printf("Time Step: %ld\n", prevTime);
             scheduleFunction[selectedAlgorithmIndex](ready_queue);
         }
+
+        // Receive any process data sent by the process_generator
+        while (msgrcv(msgQId, &msgbuf, sizeof(msgbuf.message), 0, IPC_NOWAIT) != -1)
```

Comment:
This should be handled in the "initializeProcesses" function, which is a handler to the signal sent by the generator to indicate the arrival of new processes